src/torch_models.py

```diff
@@ -146,6 +146,105 @@ def _fit(self, inputs: Array, outputs: Array) -> None:
         logging.info(f"Loaded best model with loss: {loss:.5f}")
 
 
+class ImplicitMLPRegressor(Regressor):
+    """A regressor implemented via an energy function (binary classifier).
+
+    Negative examples are generated within the class.
+
+    Inference is currently performed by sampling a fixed number of possible
+    inputs and returning the sample that has the highest probability of
+    classifying to 1, under the learned classifier. Other inference methods are
```

Comment:
is this just the input that gets the highest energy?

Reply:
yeah